queue/driver_mongo.go

```diff
@@ -430,19 +483,48 @@ func (d *mongoDriver) Get(ctx context.Context, name string) (amboy.Job, error) {
 	return output, nil
 }
 
+func (d *mongoDriver) GetAttempt(ctx context.Context, name string, attempt int) (amboy.RetryableJob, error) {
+	matchIDAndAttempt := bson.M{
+		"retry_info.base_job_id":     name,
+		"retry_info.current_attempt": attempt,
+	}
+	res := d.getCollection().FindOne(ctx, matchIDAndAttempt)
+	if err := res.Err(); err != nil {
+		return nil, errors.Wrapf(err, "GET problem fetching '%s'", name)
+	}
+
+	ji := &registry.JobInterchange{}
+	if err := res.Decode(ji); err != nil {
+		return nil, errors.Wrapf(err, "GET problem decoding '%s'", name)
+	}
+
```

Comment:
Should this call `d.removeMetadata`?

Reply:
?

Reply:
Same thing as above - since this is only used to return the Job object, it doesn't need to JobInterchange name at all.